src/crwimage_int.cpp

```diff
@@ -607,24 +607,19 @@ void CiffDirectory::doRemove(CrwDirs& crwDirs, uint16_t crwTagId) {
     auto [dir, _] = crwDirs.top();
     crwDirs.pop();
     // Find the directory
-    for (auto i = components_.begin(); i != components_.end(); ++i) {
-      if ((*i)->tag() == dir) {
-        // Recursive call to next lower level directory
-        (*i)->remove(crwDirs, crwTagId);
-        if ((*i)->empty())
-          components_.erase(i);
-        break;
-      }
+    auto it = std::find_if(components_.begin(), components_.end(), [=](auto&& c) { return c->tag() == dir; });
```

Comment:
Why `auto&& c`? There's no `std::move` here so `const auto&` would seem more natural to me.

Reply:
auto&& implicitly converts to the latter. It's shorter too.

Reply:
I'd prefer `const auto&`. It makes it clearer that this is a read-only operation.

Reply:
Yes, I also prefer the `const auto&` notation. It makes code more readable in my opinion and the reader does not need to think as much when reading that line